src/library/basesqltablemodel.cpp

```diff
@@ -280,13 +280,71 @@ void BaseSqlTableModel::select() {
         qDebug() << "Rows actually received:" << rowInfos.size();
     }
 
+    if (!m_sortedTracksIndexes.isEmpty()) {
```

Comment:
Can we get rid of this whole block, if we just use a vector of tracks? 
Removed tracks are simply missing and skipped when sorting.
New tracks are missing and can be sorted in by the code for dirty tracks. 
For a visual feedback it would be nice to either scroll to the dropped track or place it just where it was dropped, which requires even more refactoring.

Reply:
I think it would be possible, I'll take a look at it.